lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -869,6 +889,95 @@ class ExternalFunctionDefinitionsElimination : FunctionLivenessComputation {
   }
 };
 
+/// A helper class to mark all alive functions as anchors.
+class AliveFunctionsMarker : FunctionLivenessComputation {
```

Comment:
We need adding new 'markers' and phases to dead function elimination. We should sit down soon and try to figure out what the pass is actually doing and solve the problem holistically, instead of continuing to pile on more complexity here.